### PR TITLE
Add HttpClientHandler tests for server certificates

### PR DESCRIPTION
**Pull request by stephentoub**
Based in part on the ones for WinHttpHandler.  Right now these tests are only included for Unix.

cc: @davidsh, @bartonjs

### CODE REVIEW

**Comment by bartonjs** [comment]
The tests (and product fix) LGTM.

The failures indicate a) that we don't have the curl I thought we did on CI; and that b) we might need to be resilient against that in the tests for people playing the home game.

**Comment by stephentoub** [comment]
I've added a commit that updates the shim to get more details on the libcurl build being used, using that info to provide better error messages, to enable better tracing for diagnostics, and also to influence the tests.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs [hunk 1]
```diff
@@ -152,5 +152,12 @@ public static IEnumerable<object[]> UseCallback_ValidCertificate_ExpectedValuesD
                 Assert.True(callbackCalled);
             }
         }
+
+        private static bool BackendSupportsCustomCertificateHandling =>
+            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ||
+            (CurlSslVersionDescription()?.StartsWith("OpenSSL") ?? false);
```
Comment (bartonjs) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: Is this safe?  I don't know how late the JIT guarantees extern method binding.  If it's "not until first called", then yay.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs [hunk 2]
```diff
@@ -152,5 +152,12 @@ public static IEnumerable<object[]> UseCallback_ValidCertificate_ExpectedValuesD
                 Assert.True(callbackCalled);
             }
         }
+
+        private static bool BackendSupportsCustomCertificateHandling =>
```
Comment (bartonjs) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: How about a negative conditional?  To see that it's throwing the exceptions?

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: It should be fine.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs]: Added.

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please
Test Innerloop OSX Release Build and Test please

(build machines went offline during build)

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Pull request merged**